### ISSUE TITLE
Dependency markers in pyproject.toml aren't used 

### ISSUE THREAD

**Issue by bouk** [nix-community/poetry2nix#857]
### Describe the issue

I have a broken test case here: https://github.com/bouk/poetry2nix/commit/3d7d6476568d6830b441c191f8ae70a6f8afd9eb

Basically, I'm trying to use system markers to install a specific version on a specific platform. These markers aren't emitted into the lockfile (any more?) by poetry so poetry2nix doesn't use them.

I _guess_ poetry itself looks at pyproject.toml to figure out what version to install.

This seems relevant: https://github.com/python-poetry/poetry/issues/3959

### Resolution

Look at pyproject.toml when processing the lockfile to determine whether a dependency needs to be excluded

**Comment by adisbladis** [comment]
Indeed we aren't using the markers from `pyproject.toml`. We have all the logic implemented to do so though (markers are also used in poetry.lock where we _do_ check them), so adding support in pyproject.toml shouldn't be too hard.

To whomever decides to tackle this: The support is implemented in `pep508.nix` in the function `mkEvalPep508`. You can look at how this is used in `default.nix`.

**Comment by adisbladis** [comment]
> And note that to do it properly you need to implement the pubgrub and walk the dependency tree from the pyproject.toml to figure out what packages are actually required.

No, this is exactly the same pep 508 marker evaluation as we already do for poetry.lock, just in a different spot.

**Comment by bouk** [comment]
Yeah it needs to use that evaluator to determine which top-level packages need to be installed, but poetry uses the [solver](https://github.com/python-poetry/poetry/blob/master/src/poetry/puzzle/solver.py) also during installation to figure out which packages in the lock file actually need to get installed. For example, if you have markers to select a package only on windows, then that package might also have transitive dependencies that should only get installed on windows.
